### PR TITLE
Add tests for ByteArrayExtensions

### LINKED ISSUE #4: Cover ByteArrayExtensions with unit tests

### PR DESCRIPTION
**Pull request by M-Zuber**
Fixed up some bugs found while testing.
Added UTF32 Big Endian
Closes #4

### CODE REVIEW

**Review by vladislav-karamfilov — COMMENTED** [review]
[Using the file system for unit tests is not cool](http://www.artima.com/weblogs/viewpost.jsp?thread=126923). In your case a better approach would be to use `Encoding.<SomeEncoding>.GetBytes()` method for getting the bytes of a string.

**Comment by M-Zuber** [comment]
Arguable, but you have a point. 
Will verify in the morning that `...GetBytes()` returns the proper encoding markers also, to make sure it is testing a as close to real life situation as possible. 
If so will gladly update the PR. 

While I have your attention, any thoughts on how to identify a UTF16 file with zero in first two bytes after marking as opposed to thinking it is UTF32?

**Comment by M-Zuber** [comment]
@Teodor92 fixed the issue with using the FS in the tests.

**Comment by vladislav-karamfilov** [comment]
@M-Zuber, I can't think of a better approach than the additional check `bytes.Length % 4 == 0` added to the check for UTF-32. UTF-32 encoding is 4 bytes fixed-width and UTF-16 is variable 2 or 4 bytes so the check won't catch all cases correct but the change of getting the correct encoding is higher.

PS I'm sorry for my late answer. I was very busy last week...
PS2 Good job removing the file system dependency! 👍

**Comment by M-Zuber** [comment]
@vladislav-karamfilov thanks for the tip, added a check + test for a UTF16 buffer with leading zeros.

So this should be finished and ready to merge

File: Source/MoreDotNet/Extensions/Common/ByteArrayExtensions.cs [hunk 1]
```diff
@@ -33,18 +33,24 @@ public static string GetString(this byte[] buffer)
             {
                 encoding = Encoding.UTF8;
             }
-            else if (buffer[0] == 0xfe && buffer[1] == 0xff)
+
+            // In addition to preamble check the length to help UTF16 with leading zeros be recognized properly as UTF32 is 4 bytes fixed with and UTF 16 is 2
```
Comment (vladislav-karamfilov) [inline Source/MoreDotNet/Extensions/Common/ByteArrayExtensions.cs]: Small typos:
`fixed with` -> `fixed width`
`UTF16` -> `UTF-16`
`UTF 16` -> `UTF-16`
`UTF32` -> `UTF-32`

Also UTF-16 is has variable width of 2 or 4 bytes. It's good to update the comment. 😸

**Review by vladislav-karamfilov — CHANGES_REQUESTED** [review]

**Comment by vladislav-karamfilov** [comment]
It looks good to me. Just a few improvements in a comment and I can merge it. :-)

**Comment by M-Zuber** [comment]
Comments updated. Thanks.

**Pull request merged**